packages/skills/weather_client_ledger/handlers.py

```diff
@@ -0,0 +1,230 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a scaffold of a handler."""
+import json
+import logging
+from typing import Optional, cast, List
+
+from aea.configurations.base import ProtocolId
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.fipa.message import FIPAMessage
+from aea.protocols.fipa.serialization import FIPASerializer
+from aea.protocols.oef.message import OEFMessage
+from aea.protocols.oef.models import Description
+from aea.skills.base import Handler
+from aea.decision_maker.messages.transaction import TransactionMessage
+
+MAX_PRICE = 2
+STARTING_MESSAGE_ID = 1
+STARTING_TARGET_ID = 0
+
+logger = logging.getLogger("aea.weather_client_skill")
+
+
+class FIPAHandler(Handler):
+    """This class scaffolds a handler."""
+
+    SUPPORTED_PROTOCOL = FIPAMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def __init__(self, **kwargs):
+        """Initiliase the handler."""
+        super().__init__(**kwargs)
+        self.max_price = kwargs['max_price'] if 'max_price' in kwargs.keys() else MAX_PRICE
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def handle(self, message: Message, sender: str) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :param sender: the sender
+        :return: None
+        """
+        msg_performative = FIPAMessage.Performative(message.get('performative'))
+        proposals = cast(List[Description], message.get("proposal"))
+        message_id = cast(int, message.get("message_id"))
+        dialogue_id = cast(int, message.get("dialogue_id"))
+        if msg_performative == FIPAMessage.Performative.PROPOSE:
+            if proposals is not []:
+                for item in proposals:
+                    logger.info("[{}]: received proposal={} in dialogue={}".format(self.context.agent_name, item.values,
+                                                                                   dialogue_id))
+                    if "Price" in item.values.keys():
+                        if item.values["Price"] < self.max_price:
```

Comment:
The client should also have an account on ledger and check whether it can actually afford the transaction given it's wealth. So it will only accept if the tx price < max_price and if it has enough wealth to afford price.